source/postprocess/stokes_residual.cc

```diff
@@ -107,7 +107,11 @@ namespace aspect
 #endif
 
       // If there were expensive iterations add them.
-      if (solver_control_expensive.last_step() > 0)
+#if !DEAL_II_VERSION_GTE(9,0,0)
+      if (solver_control_expensive.last_step() != 0)
+#else
+      if (solver_control_expensive.last_step() != numbers::invalid_unsigned_int)
```

Comment:
I find this `! DEAL_II_VERSION_GTE(9,0,0)` hard to read. Can you just drop the `!` and switch the two branches?

OK to merge yourself then.